jung-algorithms/src/main/java/edu/uci/ics/jung/layout/algorithms/FRBHVisitorLayoutAlgorithm.java

```diff
@@ -261,11 +255,23 @@ protected void calcRepulsion(N node1) {
     }
     frNodeData.put(node1, Point.ORIGIN);
 
-    ForceObject<N> nodeForceObject = new ForceObject<>(node1, layoutModel.apply(node1));
-
-    tree.visit(repulsion_constant, nodeForceObject);
-
-    frNodeData.put(node1, fvd1.add(nodeForceObject.getForce().x, nodeForceObject.getForce().y));
+    ForceObject<N> nodeForceObject =
+        new ForceObject(node1, layoutModel.apply(node1)) {
+          @Override
+          protected void addForceFrom(ForceObject other, Optional userData) {
+            double dx = this.p.x - other.p.x;
+            double dy = this.p.y - other.p.y;
+            log.trace("dx, dy:{},{}", dx, dy);
+            double dist = Math.sqrt(dx * dx + dy * dy);
+            dist = Math.max(EPSILON, dist);
+            log.trace("dist:{}", dist);
+            double force = (repulsion_constant * repulsion_constant) / dist;
+            log.trace("force:{}", force);
+            f = f.add(force * (dx / dist), force * (dy / dist));
+          }
+        };
+    tree.visit(nodeForceObject, Optional.empty());
+    frNodeData.put(node1, nodeForceObject.f);
   }
```

Comment:
instead of leaving the BHQT classes hard-coded for FRLayout, I override a method here to do the right implementation. This let me do the same thing in SpringLayout

Reply:
Or, instead of an anonymous inner-class that 'sees' values like 'repulsion_constant', a user-data object could be constructed and sent down the visitor path instead. You may even be able to make a lambda function pointer to pass as userdata.

Reply:
I ended up taking out the userData. I tried a few things but did not find anything where I needed it as part of the solution.

Reply:
Seems like a good call.  We can always revisit if it seems warranted.